jobs/build-fcos-buildroot.Jenkinsfile

```diff
@@ -146,12 +146,16 @@ lock(resource: "build-${containername}") {
                     def arch = architecture
                     images += " --image=docker://${params.CONTAINER_REGISTRY_STAGING_REPO}:${arch}-${shortcommit}"
                 }
-                shwrap("""
-                export STORAGE_DRIVER=vfs # https://github.com/coreos/fedora-coreos-pipeline/issues/723#issuecomment-1297668507
-                cosa push-container-manifest \
-                    --auth=\$REGISTRY_SECRET --tag ${gitref} \
-                    --repo ${params.CONTAINER_REGISTRY_REPO} ${images}
-                """)
+                // arbitrarily selecting the x86_64 builder; we don't run this
+                // locally because podman wants user namespacing (yes, even just
+                // to push a manifest...)
+                pipeutils.withPodmanRemoteArchBuilder(arch: "x86_64") {
+                    shwrap("""
+                        cosa push-container-manifest \
+                            --auth=\$REGISTRY_SECRET --tag ${gitref} \
```

Comment:
I'm guessing `$REGISTRY_SECRET` here is the actual value and isn't a path to a file because I don't see where that is getting synced to the remote builder.

Reply:
Yeah it's weird. I was confused by that too but somehow it works.

Reply:
OK so I think I understand this. This works in the `build-*` jobs because we're not using the remote session stuff there, only setting the `CONTAINER_HOST` bits and using the cosa from the pod. The cosa commands call podman, which takes care of proxying through `REGISTRY_AUTH_FILE`.

In the `release` job, I think it worked in testing because I initially tested with `cosa supermin-run` and it did push. And then when I tested with the remote builder approach, it no-oped on the actual push phase because nothing changed so didn't actually hit a credentials issue, but I think it would've failed. I pushed an update but will test it either later today or tomorrow morning.